src/main/java/uk/gov/hmcts/darts/audio/mapper/GetTransformedMediaDetailsMapper.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.darts.audio.mapper;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.audio.model.SearchTransformedMediaResponse;
+import uk.gov.hmcts.darts.audio.model.SearchTransformedMediaResponseCase;
+import uk.gov.hmcts.darts.audio.model.SearchTransformedMediaResponseCourthouse;
+import uk.gov.hmcts.darts.audio.model.SearchTransformedMediaResponseHearing;
+import uk.gov.hmcts.darts.audio.model.SearchTransformedMediaResponseMediaRequest;
+import uk.gov.hmcts.darts.common.entity.TransformedMediaEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class GetTransformedMediaDetailsMapper {
+    public List<SearchTransformedMediaResponse> mapSearchResults(List<TransformedMediaEntity> entityList) {
+        List<SearchTransformedMediaResponse> mappedDetails = new ArrayList<>();
+        for (TransformedMediaEntity entity : entityList) {
+            SearchTransformedMediaResponse transformedMediaDetails = new SearchTransformedMediaResponse();
+            transformedMediaDetails.setId(entity.getId());
+            transformedMediaDetails.setFileName(entity.getOutputFilename());
+
+            if (entity.getOutputFormat() != null) {
+                transformedMediaDetails.setFileFormat(entity.getOutputFormat().name());
+            }
+
+            transformedMediaDetails.setLastAccessedAt(entity.getLastAccessed());
+            transformedMediaDetails.setFileSizeBytes(entity.getOutputFilesize());
+
+            if (entity.getMediaRequest() != null) {
+                SearchTransformedMediaResponseMediaRequest mediaRequest = new SearchTransformedMediaResponseMediaRequest();
+                mediaRequest.setId(entity.getMediaRequest().getId());
+                mediaRequest.setRequestedAt(entity.getMediaRequest().getCreatedDateTime());
+                mediaRequest.setOwnerUserId(entity.getMediaRequest().getCurrentOwner().getId());
+                mediaRequest.setRequestedByUserId(entity.getMediaRequest().getRequestor().getId());
+                transformedMediaDetails.setMediaRequest(mediaRequest);
+            }
+
+            if (entity.getMediaRequest() != null && entity.getMediaRequest().getHearing() != null
+                && entity.getMediaRequest().getHearing().getCourtCase() != null) {
```

Comment:
if entity.getMediaRequest().getHearing()!=null, i'm not sure its possible for its courtCase to be null as the database has a non-null constraint.

Reply:
yeah just being defensive i can remove